### ISSUE TITLE
$accumulator support (MongoDB 4.4)

### ISSUE THREAD

**Issue by dimaip** [Automattic/mongoose#9364]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

When using the [example from the docs](https://docs.mongodb.com/manual/reference/operator/aggregation/accumulator/) on `$accumulator` I get a weird error `$accumulator missing required argument 'init'` even though I clearly define `init`.

**If the current behavior is a bug, please provide the steps to reproduce.**

https://gist.github.com/dimaip/17365161ad6e6615cd50bda4b75ad93c

**What is the expected behavior?**

It should work as the docs describe

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

Mongo 4.4, Mongoose 5.10.0, Node 10.18.1

**Comment by vkarpov15** [comment]
Unfortunately, the MongoDB driver doesn't serialize functions by default, which is a problem for `$accumulator`. You have two options currently:

1) Use template strings to make your functions into strings:

```javascript
$accumulator: {
  init: `function() {
    return { count: 0, sum: 0 }
  }`
  // ...
```

2) Set the `serializeFunctions` option:

```javascript
Model.aggregate([/*...*/]).option({ serializeFunctions: true });
```

We'll add a fix for this in Mongoose 5.10.2 so Mongoose handles this case automatically :+1:

**Comment by dimaip** [comment]
Thanks so much for the response and the fix!
